From 0a754547775c6fa99d339a18e07fc9bca013b3ca Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Fri, 17 Jan 2014 17:07:51 +0100
Subject: [PATCH 01/11] Revert "qdev-monitor: Fix crash when device_add is called with abstract driver"

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1389978479-30595-2-git-send-email-armbru@redhat.com>
Patchwork-id: 56789
O-Subject: [PATCH 7.0 qemu-kvm 1/9] Revert "qdev-monitor: Fix crash when device_add is called with abstract driver"
Bugzilla: 669524
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>

This reverts commit 3e97a22728afed3c77140486ec3863eba941535b.

Reverting because it makes -device nonexistant crash.  Incorrect
conflict resolution.  I'm going to reapply the upstream commit in a
place where it applies cleanly, and actually works.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 qdev-monitor.c | 6 ------
 1 file changed, 6 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qdev-monitor.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/qdev-monitor.c b/qdev-monitor.c
index 979502e..f78ff64 100644
--- a/qdev-monitor.c
+++ b/qdev-monitor.c
@@ -477,12 +477,6 @@ DeviceState *qdev_device_add(QemuOpts *opts)
         }
     }
 
-    if (object_class_is_abstract(obj)) {
-        qerror_report(QERR_INVALID_PARAMETER_VALUE, "driver",
-                      "non-abstract device type");
-        return NULL;
-    }
-
     k = DEVICE_CLASS(obj);
 
     if (!k || k->cannot_instantiate_with_device_add_yet) {
-- 
1.7.1

